crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -462,14 +462,14 @@ impl<T: TransactionOrdering> TxPool<T> {
         &mut self,
         changed_senders: HashMap<SenderId, SenderInfo>,
     ) -> UpdateOutcome<T::Transaction> {
-        // track changed accounts
-        self.sender_info.extend(changed_senders.clone());
         // Apply the state changes to the total set of transactions which triggers sub-pool updates.
-        let updates = self.all_transactions.update(changed_senders);
+        let updates = self.all_transactions.update(&changed_senders);
         // Process the sub-pool updates
         let update = self.process_updates(updates);
         // update the metrics after the update
         self.update_size_metrics();
+        // track changed accounts
+        self.sender_info.extend(changed_senders);
```

Comment:
can we do this like #13194 (before `self.process_updates`) then it's obvious that this order is safe.

Reply:
Done, thanks!